ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -121,27 +123,42 @@ func (r *RayClusterReconciler) eventReconcile(request ctrl.Request, event *corev
 	// we only care about pod events
 	if event.InvolvedObject.Kind != "Pod" || event.Type != "Warning" || event.Reason != "Unhealthy" ||
 		!strings.Contains(event.Message, "Readiness probe failed") {
+		// This is not supposed to happen
 		return ctrl.Result{}, nil
 	}
 
 	_ = r.Log.WithValues("event", request.NamespacedName)
 	r.Log.Info("reconcile RayCluster Event", "event name", request.Name)
 
-	if err := r.List(context.TODO(), &pods,
-		client.InNamespace(event.InvolvedObject.Namespace)); err != nil {
+	options := []client.ListOption{
+		client.MatchingFields(map[string]string{"metadata.name": event.InvolvedObject.Name}),
+		client.InNamespace(event.InvolvedObject.Namespace),
+		client.MatchingLabels(map[string]string{common.RayNodeLabelKey: "yes"}),
+	}
+	r.Log.Info("list pods", "options", options)
+
+	if err := r.List(context.TODO(), &pods, options...); err != nil {
 		return ctrl.Result{}, err
 	}
 
-	for _, item := range pods.Items {
-		if item.Name == event.InvolvedObject.Name {
-			if unhealthyPod != nil {
-				return ctrl.Result{}, fmt.Errorf("duplicate pod found")
+	if len(pods.Items) == 0 {
+		r.Log.Error(fmt.Errorf("no pod found for event"), "no pod found for event",
+			"event", event)
+		return ctrl.Result{}, nil
+	} else if len(pods.Items) > 1 {
+		// This happens when we use fake client
```

Comment:
this happens during test which we do not have the server side filtering available.

Reply:
Would you mind explaining it more? In my understanding, it is impossible to have two Pods with the same name in an actual situation? Why do we need to test this case? Do I miss anything? Thanks!

Reply:
This is happening during test in which we are using fakeclient which does not support these different kinds of filters applied in the ListOption. If we don't have this, the test will fail which is not because of a real failure.